### PR TITLE
Drop usage of deprecated functions

### PR DESCRIPTION
**Pull request by acumino**
**How to categorize this PR?**

/area open-source
/kind cleanup

**What this PR does / why we need it**:
This PR does following changes - 
- Drop usage of `strings.Title` and use recommended `cases.Title`.
- Drop usage of `Stream` and use `StreamWithContext`.

**Which issue(s) this PR fixes**:
Fixes #

**Special notes for your reviewer**:

**Release note**:

```other operator
NONE
```

### CODE REVIEW

File: pkg/component/etcd/monitoring.go [hunk 1]
```diff
@@ -401,9 +403,12 @@ func (e *etcd) AlertingRules() (map[string]string, error) {
 		etcdReplicas = *e.values.Replicas
 	}
 
+	// Straightforward approach
+	caser := cases.Title(language.English)
+
 	if err := monitoringAlertingRulesTemplate.Execute(&alertingRules, map[string]interface{}{
 		"role":               e.values.Role,
-		"Role":               strings.Title(e.values.Role),
+		"Role":               caser.String(e.values.Role),
```
Comment (shafeeqes) [inline pkg/component/etcd/monitoring.go]: Use `cases.Title(language.English).String(e.values.Role)` ?

Comment (acumino) [reply pkg/component/etcd/monitoring.go]: Updated in commit 5aa055c88d38c759787cefd812e3ac0e4c77cb36

**Review by shafeeqes — APPROVED** [review]
/lgtm

**Pull request merged**